### PR TITLE
Update to heapster v0.14.0

### PR DESCRIPTION
**Pull request by vishh**

### CODE REVIEW

**Comment by vishh** [comment]
$ hack/ginkgo-e2e.sh --ginkgo.focus="Monitoring"
Setting up for KUBERNETES_PROVIDER="gce".
Project: vishnuk-cloud
Zone: us-central1-b
>>> testContext.KubeConfig: /usr/local/google/home/vishnuk/.kube/config
INFO: Waiting up to 10m0s for all pods (need at least 0) in namespace 'default' to be running and ready
INFO: 8 / 8 pods in namespace 'default' are running and ready (5 seconds elapsed)
Running Suite: Kubernetes e2e suite
===================================
Random Seed: 1433972365 - Will randomize all specs
Will run 1 of 73 specs

SSSSSSSSSSSSSSSSS
------------------------------
Monitoring 
  should verify monitoring pods and all cluster nodes are available on influxdb using heapster.
  /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/monitoring.go:51
[BeforeEach] Monitoring
  /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/monitoring.go:42
>>> testContext.KubeConfig: /usr/local/google/home/vishnuk/.kube/config
[It] should verify monitoring pods and all cluster nodes are available on influxdb using heapster.
  /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/monitoring.go:51
•SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS
Ran 1 of 73 Specs in 0.933 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 72 Skipped PASS

**Comment by vishh** [comment]
cc @dchen1107

**Comment by saad-ali** [comment]
LGTM

File: cluster/addons/cluster-monitoring/google/heapster-service.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+kind: "Service"
+apiVersion: "v1beta3"
```
Comment (lavalamp) [inline cluster/addons/cluster-monitoring/google/heapster-service.yaml]: Please use v1

File: cluster/addons/cluster-monitoring/influxdb/heapster-service.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+kind: "Service"
+apiVersion: "v1beta3"
```
Comment (lavalamp) [inline cluster/addons/cluster-monitoring/influxdb/heapster-service.yaml]: v1

**Comment by vishh** [comment]
@lavalamp: Can the update to 'v1' be done in a separate PR for all the config files?

**Comment by dchen1107** [comment]
I suggested not. What is blocking you to change those to v1 instead of v1beta3?

**Comment by vishh** [comment]
Ok! All the monitoring configs have been updated. PTAL

**Comment by vishh** [comment]
E2e passed again with updated configs:
```shell
Monitoring 
  should verify monitoring pods and all cluster nodes are available on influxdb using heapster.
  /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/monitoring.go:51
[BeforeEach] Monitoring
  /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/monitoring.go:42
>>> testContext.KubeConfig: /usr/local/google/home/vishnuk/.kube/config
[It] should verify monitoring pods and all cluster nodes are available on influxdb using heapster.
  /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/monitoring.go:51
•SSSSSSSSSSSSSSSSSSSSS
Ran 1 of 73 Specs in 0.857 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 72 Skipped PASS

Ginkgo ran 1 suite in 6.28581304s
Test Suite Passed
```

**Comment by lavalamp** [comment]
This LGTM now, but I wonder why the "-v2" is needed in the name--is the label not sufficient?

**Comment by vishh** [comment]
@lavalamp: The version labels were meant for rolling updates of addons during cluster upgrades. cc @satnam6502 to make sure I am satisfying the original design goals.

**Comment by dchen1107** [comment]
LGTM thanks!

**Comment by satnam6502** [comment]
Looks great to me.

**Comment by vishh** [comment]
@ArtfulCoder: This PR needs to go in soon. Can you prioritize this?

**Pull request merged**